The second commit simplifies handling of such exception by just letting the client fail without the wrapping into a ProblemDetail as well as not wrapping ClientExceptions into another ClientException as currently happening in `HttpZeebeFuture.unwrapExecutionException`

```
Exception in thread "main" io.camunda.zeebe.client.api.command.ClientException: io.camunda.zeebe.client.api.command.MalformedResponseException: Expected to receive a response body, but got a problem: class ProblemDetail {
    type: about:blank
    title: Unexpected server response
    status: 500
    detail: {"jobs":[{"type":"check-inventory","processDefinitionId":"process1","processDefinitionVersion":1,"elementId":"Activity_0tw2fu0","customHeaders":{},"worker":"default","retries":3,"deadline":1760695339434,"variables":{"item":"special widget"},"tenantId":"<default>","jobKey":"2251799813815741","processInstanceKey":"2251799813815735","processDefinitionKey":"2251799813815732","elementInstanceKey":"2251799813815740","kind":"BPMN_ELEMENT","listenerEventType":"UNSPECIFIED","tags":[]}]}
    instance: null
}
	at io.camunda.zeebe.client.impl.http.HttpZeebeFuture.unwrapExecutionException(HttpZeebeFuture.java:86)
	at io.camunda.zeebe.client.impl.http.HttpZeebeFuture.join(HttpZeebeFuture.java:42)
	at io.camunda.zeebe.client.TestWorker.main(TestWorker.java:65)
Caused by: io.camunda.zeebe.client.api.command.MalformedResponseException: Expected to receive a response body, but got a problem: class ProblemDetail {
    type: about:blank
    title: Unexpected server response
    status: 500
    detail: {"jobs":[{"type":"check-inventory","processDefinitionId":"process1","processDefinitionVersion":1,"elementId":"Activity_0tw2fu0","customHeaders":{},"worker":"default","retries":3,"deadline":1760695339434,"variables":{"item":"special widget"},"tenantId":"<default>","jobKey":"2251799813815741","processInstanceKey":"2251799813815735","processDefinitionKey":"2251799813815732","elementInstanceKey":"2251799813815740","kind":"BPMN_ELEMENT","listenerEventType":"UNSPECIFIED","tags":[]}]}
    instance: null
}
	at io.camunda.zeebe.client.impl.http.ApiCallback.handleSuccessResponse(ApiCallback.java:139)
```

After:
```
Exception in thread "main" io.camunda.zeebe.client.api.command.ClientException: com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "kind" (class io.camunda.zeebe.client.protocol.rest.ActivatedJob), not marked as ignorable (14 known properties: "jobKey", "tenantId", "retries", "elementInstanceKey", "customHeaders", "processDefinitionVersion", "elementId", "deadline", "type", "processInstanceKey", "processDefinitionKey", "worker", "variables", "processDefinitionId"])
 at [Source: UNKNOWN; byte offset: #UNKNOWN] (through reference chain: io.camunda.zeebe.client.protocol.rest.JobActivationResponse["jobs"]->java.util.ArrayList[0]->io.camunda.zeebe.client.protocol.rest.ActivatedJob["kind"])
	at io.camunda.zeebe.client.impl.http.ApiCallback.failed(ApiCallback.java:73)
	at org.apache.hc.core5.concurrent.BasicFuture.failed(BasicFuture.java:166)
	at org.apache.hc.core5.concurrent.ComplexFuture.failed(ComplexFuture.java:79)
	at org.apache.hc.client5.http.impl.async.InternalAbstractHttpAsyncClient$2.failed(InternalAbstractHttpAsyncClient.java:364)
	at org.apache.hc.client5.http.impl.async.AsyncRedirectExec$1.failed(AsyncRedirectExec.java:246)
	at org.apache.hc.client5.http.impl.async.AsyncHttpRequestRetryExec$1.failed(AsyncHttpRequestRetryExec.java:195)
	at org.apache.hc.client5.http.impl.async.AsyncProtocolExec$1.failed(AsyncProtocolExec.java:295)
	at org.apache.hc.client5.http.impl.async.HttpAsyncMainClientExec$1.failed(HttpAsyncMainClientExec.java:135)
```

--- a/clients/java/src/main/java/io/camunda/zeebe/client/impl/http/HttpClientFactory.java
+++ b/clients/java/src/main/java/io/camunda/zeebe/client/impl/http/HttpClientFactory.java
@@ -15,6 +15,7 @@
  */
 package io.camunda.zeebe.client.impl.http;
 
+import com.fasterxml.jackson.databind.DeserializationFeature;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import io.camunda.zeebe.client.CredentialsProvider;
 import io.camunda.zeebe.client.ZeebeClientConfiguration;
@@ -69,7 +70,8 @@ public class HttpClientFactory {
   /** The versioned base REST API context path the client uses for requests */
   public static final String REST_API_PATH = "/v2";
 
-  private static final ObjectMapper JSON_MAPPER = new ObjectMapper();
+  private static final ObjectMapper JSON_MAPPER =
+      new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
 
   private final ZeebeClientConfiguration config;
 

--- a/clients/java/src/test/java/io/camunda/zeebe/client/ExceptionHandlingRestTest.java
+++ b/clients/java/src/test/java/io/camunda/zeebe/client/ExceptionHandlingRestTest.java
@@ -15,9 +15,11 @@
  */
 package io.camunda.zeebe.client;
 
+import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
 import io.camunda.zeebe.client.api.command.ProblemException;
+import io.camunda.zeebe.client.api.response.ActivateJobsResponse;
 import io.camunda.zeebe.client.protocol.rest.ProblemDetail;
 import io.camunda.zeebe.client.util.ClientRestTest;
 import io.camunda.zeebe.client.util.RestGatewayPaths;
@@ -51,5 +53,21 @@ public final class ExceptionHandlingRestTest extends ClientRestTest {
             () -> client.newTopologyRequest().useRest().send().join(1L, TimeUnit.SECONDS))
         .isInstanceOf(ProblemException.class)
         .hasMessageContaining("Invalid request");
+  }
+
+  // regression test for https://github.com/camunda/camunda/issues/39675
+  @Test
+  public void shouldIgnoreUnknownProperties() {
+    // given / a response with an unknown property "kind" in a job
+    gatewayService.onActivateJobsRequest(
+        "{\"jobs\": [{ \"type\": \"type\", \"kind\": \"BPMN_ELEMENT\"}]}");
+
+    // when / then
+    final ActivateJobsResponse activateJobsResponse =
+        client.newActivateJobsCommand().jobType("type").maxJobsToActivate(1).send().join();
+    assertThat(activateJobsResponse).isNotNull();
+    assertThat(activateJobsResponse.getJobs())
+        .hasSize(1)
+        .allSatisfy(job -> assertThat(job.getType()).isEqualTo("type"));
   }
 }

--- a/clients/java/src/test/java/io/camunda/zeebe/client/util/RestGatewayService.java
+++ b/clients/java/src/test/java/io/camunda/zeebe/client/util/RestGatewayService.java
@@ -48,6 +48,20 @@ public class RestGatewayService {
   /**
    * Register the given response for job activation requests.
    *
+   * @param jobActivationResponseJsonString the response to provide upon a job activation request as
+   *     json string
+   */
+  public void onActivateJobsRequest(final String jobActivationResponseJsonString) {
+    mockInfo
+        .getWireMock()
+        .register(
+            WireMock.post(RestGatewayPaths.getJobActivationUrl())
+                .willReturn(WireMock.okJson(jobActivationResponseJsonString)));
+  }
+
+  /**
+   * Register the given response for job activation requests.
+   *
    * @param jobActivationResponse the response to provide upon a job activation request
    */
   public void onActivateJobsRequest(final JobActivationResponse jobActivationResponse) {

